python/kserve/kserve/model_repository.py

```diff
@@ -48,11 +48,11 @@ def get_model(self, name: str) -> Optional[BaseKServeModel]:
     def get_models(self) -> Dict[str, BaseKServeModel]:
         return self.models
 
-    def is_model_ready(self, name: str):
+    async def is_model_ready(self, name: str):
         model = self.get_model(name)
         if not model:
             return False
-        return model.healthy()
+        return await model.healthy()
```

Comment:
maybe we need to define low timeout here to avoid a high number of threads in case the health check gets stuck or too slow to answer, e.g. if the vllm server is very busy.